src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1025,6 +1026,12 @@ struct SmartAction
             uint32 wp6;
         } closestWaypointFromList;
 
+        struct
+        {
+            uint32 sound[5];
```

Comment:
I didnt mean to say it was wrong, I meant that it could be improved so that it would use only one definition to define the amount of iterations and elements in the array/container. Using the same method for both cases or using some constant to define the size for both or eliminating the need for the amount of elements in the loop, with a range based for loop for example, would do the job.